### PR TITLE
Migrate CircleCI unit test workflow to Github  (contrib)

### PR DESCRIPTION
**Pull request by shovnik**
## Which problem is this PR solving?
- This addresses [issue](https://github.com/open-telemetry/opentelemetry-js/issues/1259#event-4038172785) by migrating the only existing opentelemetry-js-contrib workflow from CircleCI to GitHub Actions (config.yml) 

## Short description of the changes
- Migrate node-unit-tests and browser-unit-tests contained in config.yml file to GitHub Actions
- Change lint workflow trigger to be on pull_request and push as well to avoid overlapping triggers
- Minor unrelated lint fix so that the lint test stops failing for every PR on the contrib repo

## Migration Plan
I suggest having CircleCI and GitHub Action jobs run in parallel for a few weeks. After the GitHub Action jobs are running fine for a week or so and then remove the CircleCI workflows from config.yml

cc - @alolita

### CODE REVIEW

**Comment by shovnik** [comment]
I deleted the CircleCI workflow file, but to complete the migration a maintainer needs to remove the CircleCi webhook in the repository settings before merging this PR.

**Review by dyladan — APPROVED** [review]

**Review by vmarchaud — APPROVED** [review]

**Pull request closed without merging**
